### PR TITLE
Fix Issue #997

### PR DESCRIPTION
**Pull request by feiyun0112**
Get name from ColumnNameAttribute

### CODE REVIEW

File: src/Microsoft.ML/Data/TextLoader.cs [hunk 1]
```diff
@@ -95,7 +95,8 @@ public sealed partial class TextLoader
                     throw Contracts.Except($"{mappingAttr.Ordinal} contains invalid characters. " +
                         $"Valid characters are 0-9, *, - and ~");
 
-                var name = mappingAttr.Name ?? memberInfo.Name;
+                var mappingNameAttr = memberInfo.GetCustomAttribute<ColumnNameAttribute>();
+                var name = mappingAttr.Name ?? mappingNameAttr?.Name ?? memberInfo.Name;
```
Comment (TomFinley) [inline src/Microsoft.ML/Data/TextLoader.cs]: >mappingAttr.Name ?? mappingNameAttr?.Name [](start = 27, length = 41)

Thanks @feiyun0112. I wonder if a slightly different way of doing things might be appropriate. This currently tolerates the situation where .

Previously we have one attribute `ColumnName` to be used in the prediction engine, and another attribute `Column` to be used with specifying things for the text loader. Making the text loader possibly use the same attribute as that used in the pipeline engine seems harmless enough.

However, I wonder, should we if both are specified *fail*? It seems very undesirable that a user should be able to specify names (possibly different) in two distinct locations, and that's just a-OK from the system's perspective.

Even *if* we wanted to tolerate this situation which I don't think we do, I'd rather that the `ColumnName` attribute take precedence... that is, I might prefer `mappingNameAttr?.Name ?? mappingAttr.Name ?? memberInfo.Name`, that is, reverse the first two.

Comment (feiyun0112) [reply src/Microsoft.ML/Data/TextLoader.cs]: I used same precedence as [Microsoft.ML.Api/SchemaDefinition.cs ](https://github.com/dotnet/machinelearning/blob/6f4c8c5c3d9cfe1fcb943c597f793155b0d092f8/src/Microsoft.ML.Api/SchemaDefinition.cs)  396 line
```
var mappingAttr = memberInfo.GetCustomAttribute<ColumnAttribute>();
var mappingNameAttr = memberInfo.GetCustomAttribute<ColumnNameAttribute>();
string name = mappingAttr?.Name ?? mappingNameAttr?.Name ?? memberInfo.Name;
```

Comment (TomFinley) [reply src/Microsoft.ML/Data/TextLoader.cs]: Hmmm. Well I guess being consistent with that is fine, all right.

**Comment by justinormont** [comment]
This PR could have used  a better name.

**Comment by TomFinley** [comment]
@justinormont yes... fortunately the commit message was descriptive.

**Pull request merged**